src/main/java/seedu/address/logic/commands/MarkAttendanceCommand.java

```diff
@@ -0,0 +1,138 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DATE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TELEGRAM;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_PERSONS;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.attendance.Attendance;
+import seedu.address.model.person.Email;
+import seedu.address.model.person.Name;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.Phone;
+import seedu.address.model.person.Telegram;
+import seedu.address.model.role.Role;
+
+/**
+ * Mark the attendance of people with targeted telegram handle.
+ */
+public class MarkAttendanceCommand extends AttendanceMarkingCommand {
+    public static final String COMMAND_WORD = "mark";
+    public static final String COMMAND_ALIAS = "m";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Mark the attendance of people with list of input telegrams on designated date.\n"
+            + "Parameters: "
+            + PREFIX_TELEGRAM + "TELEGRAM "
+            + PREFIX_DATE + "DATE "
+            + "Example: " + COMMAND_WORD + " " + PREFIX_TELEGRAM + "johndoe "
```

Comment:
1. maybe can add '\n' to show the example usage on a new line so that it aligns with other command usages.
2. maybe can use 'alexYeoh' or 'berniceYu' for the example instead, since they will exist in the sample addressbook

Reply:
Thanks, changed already.